src/components/form/carForm.tsx

```diff
@@ -1,115 +1,115 @@
-
-import { validateCar } from '../../lib/utils'
+import { useForm, SubmitHandler, Controller, Control, FieldErrors } from "react-hook-form";
 
 interface CarFormProps { addCar: (car: Car) => void }
 
 interface FormFieldsProps {
-    car: Car,
-    handleChange: (e: ChangeEvent<HTMLInputElement>) => void,
-    errors: FormErrors
+    control: Control<Car>
+    errors: FieldErrors
 }
 
-export function FormFields({ car, handleChange, errors } : FormFieldsProps) {
+export function FormFields({control, errors} : FormFieldsProps) {
+
     return (
         <>
-            <TextField
-                sx={{marginY: 1}}
-                label='Maker'
+            <Controller
                 name='maker'
-                select
-                value={car.maker}
-                onChange={handleChange}
-                error={errors.maker !== ''}
-                helperText={errors.maker}
-                fullWidth>
-                    <MenuItem value="">Select Maker</MenuItem>
-                    <MenuItem value="Toyota">Toyota</MenuItem>
-                    <MenuItem value="Honda">Honda</MenuItem>
-                    <MenuItem value="Ford">Ford</MenuItem>
-            </TextField>
-            <TextField
-                sx={{marginY: 1}}
-                label='Model'
+                control={control}
+                rules={{required: 'Maker is required'}}
+                render={({field}) => (
+                    <TextField
+                        sx={{marginY: 1}}
+                        label='Maker'
+                        select
+                        fullWidth
+                        error={errors.maker !== undefined}
+                        helperText={errors.maker?.message?.toString()}
+                        {...field}>
+                            <MenuItem value="">Select Maker</MenuItem>
+                            <MenuItem value="Toyota">Toyota</MenuItem>
+                            <MenuItem value="Honda">Honda</MenuItem>
+                            <MenuItem value="Ford">Ford</MenuItem>
+                    </TextField>
+                )}
+            />
+            <Controller
                 name='model'
-                value={car.model}
-                onChange={handleChange}
-                error={errors.model !== ''}
-                helperText={errors.model}
-                fullWidth
+                control={control}
+                rules={{required: 'Model is required'}}
+                render={({field}) => (
+                    <TextField
+                        sx={{marginY: 1}}
+                        label='Model'
+                        error={errors.model !== undefined}
+                        helperText={errors.model?.message?.toString()}
+                        fullWidth
+                        {...field}
+                    />
+                )}
             />
```

Comment:
The thing that I saw in Ira PR. Could you try to use the approach with `register` instead of wrapping a textfield into controller. You will see how it reduce the code:
Pls. check the comment from Ira's PR: https://github.com/proga-react-practice/kappa-team-project/pull/3#discussion_r1598781918